source/extensions/quic_listeners/quiche/active_quic_listener.cc

```diff
@@ -112,14 +117,37 @@ void ActiveQuicListener::onData(Network::UdpRecvData& data) {
                                   /*packet_headers=*/nullptr, /*headers_length=*/0,
                                   /*owns_header_buffer*/ false);
   quic_dispatcher_->ProcessPacket(self_address, peer_address, packet);
+
+  if (!schedule_process_buffered_enabled_) {
+    // On event loop runs where this listener reads or has packets posted to it, process buffered
+    // packets on the next loop. This ensures that the limit of processed client hellos per loop is
+    // honored, and that buffered client hellos are processed in a timely fashion.
+    schedule_process_buffered_->scheduleCallbackNextIteration();
+    schedule_process_buffered_enabled_ = true;
+  }
 }
 
-void ActiveQuicListener::onReadReady() {
+void ActiveQuicListener::onReadReady() {}
+
+void ActiveQuicListener::processBuffered() {
+  schedule_process_buffered_enabled_ = false;
+
   if (!enabled_.enabled()) {
     ENVOY_LOG(trace, "Quic listener {}: runtime disabled", config_->name());
     return;
   }
+
+  if (quic_dispatcher_->HasChlosBuffered()) {
+    event_loop_chlo_buffered_++;
+  }
+
   quic_dispatcher_->ProcessBufferedChlos(kNumSessionsToCreatePerLoop);
+
+  // If there were more buffered than the limit, schedule again for the next event loop.
+  if (quic_dispatcher_->HasChlosBuffered()) {
```

Comment:
Oh, that's a good point; I hadn't thought of that. Let me look into it and make sure that won't cause any issues.

Reply:
Actually, it looks like `activate()` now uses `scheduleCallbackNextIteration()`, so I don't think there's much/any benefit to changing it. @antoniovicente made the changes in #11750. @antoniovicente can you comment on whether there's any perf benefit to using `FileEvent::activate()` here instead of this?

Reply:
I would think the benefit is code simplicity since you wouldn't need to keep your own SchedulableCallback to do the wakeup.  The argument for adding it temporarily is that next iteration scheduling is protected by a runtime feature.  Would anything break if the callback was scheduled in the current iteration instead?

Reply:
activate() seems to merge with incoming IO events and there is no need to add an extra callback in this class :)

Reply:
> I would think the benefit is code simplicity since you wouldn't need to keep your own SchedulableCallback to do the wakeup. The argument for adding it temporarily is that next iteration scheduling is protected by a runtime feature. Would anything break if the callback was scheduled in the current iteration instead?

Quic listener will drain the queued CHLOs in current loop instead of postpone till next loop. It will always make forward progress. I think it's okay to use `activate()`.

Reply:
> Would anything break if the callback was scheduled in the current iteration instead?

I think that in some cases, we could burst to handling 2x the configured number of CHLOs in one event loop, but in a steady state it would handle the correct number, so I think that's fine; the buffering behavior doesn't need to be perfect.